commit 8179be21e664cedb2e9d238cc2f6d04965e97275
Author: Sergey Poznyakoff <gray@gnu.org>
Date:   Thu May 11 10:18:44 2023 +0300

    configure: Include needed header for major/minor macros
    
    This helps in avoiding the warning about implicit function declaration
    which is elevated as error with newer compilers e.g. clang 16
    
    Signed-off-by: Khem Raj <raj.khem@gmail.com>

diff --git a/configure.ac b/configure.ac
index de479e7454e5d035..c60102972282b63b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -43,8 +43,22 @@ AC_TYPE_UID_T
 AC_CHECK_TYPE(gid_t, int)
 
 AC_HEADER_DIRENT
-AX_COMPILE_CHECK_RETTYPE([major], [0])
-AX_COMPILE_CHECK_RETTYPE([minor], [0])
+AX_COMPILE_CHECK_RETTYPE([major], [0], [
+#include <sys/types.h>
+#ifdef MAJOR_IN_MKDEV
+# include <sys/mkdev.h>
+#endif
+#ifdef MAJOR_IN_SYSMACROS
+# include <sys/sysmacros.h>
+#endif])
+AX_COMPILE_CHECK_RETTYPE([minor], [0], [
+#include <sys/types.h>
+#ifdef MAJOR_IN_MKDEV
+# include <sys/mkdev.h>
+#endif
+#ifdef MAJOR_IN_SYSMACROS
+# include <sys/sysmacros.h>
+#endif])
 
 AC_CHECK_FUNCS([fchmod fchown])
 # This is needed for mingw build
